src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -433,5 +434,310 @@ private static List<string> GetNamesOfUserProfiles()
+            }
+
+            Assert.False(environmentVariables.ContainsKey("NewKey99"));
+
+            //Key not found
+            Assert.Throws<KeyNotFoundException>(() =>
+            {
+                string stringout = environmentVariables["NewKey99"];
+            });            
+
+            //Exception not thrown with invalid key
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                string stringout = environmentVariables[null];
+            });
+
+            //Exception not thrown with invalid key
+            Assert.Throws<ArgumentNullException>(() => environmentVariables.Add(null, "NewValue2"));
+
+            //Invalid Key to add
+            Assert.Throws<ArgumentException>(() => environmentVariables.Add("NewKey2", "NewValue2"));
+
+            //Remove Item
+            environmentVariables.Remove("NewKey98");
+
+            //Exception not thrown with null key
+            Assert.Throws<ArgumentNullException>(() => { environmentVariables.Remove(null); });
+
+            //"Exception not thrown with null key"
+            Assert.Throws<KeyNotFoundException>(() => environmentVariables["1bB"]);
+
+            Assert.True(environmentVariables.ContainsKey("NewKey2"));
+            Assert.True(environmentVariables.ContainsValue("NewValue2"));
+
+            Assert.False(environmentVariables.ContainsValue("newvalue2"));
+            /*Assert.False(environmentVariables.ContainsKey("newkey2"));*/
+
+            //Use KeyValuePair Enumerator
+            var x = environmentVariables.GetEnumerator() as IEnumerator<KeyValuePair<string, string>>;
+            x.MoveNext();
+            var y1 = x.Current;
+            Assert.Equal("NewKey NewValue", y1.Key + " " + y1.Value);
+            x.MoveNext();
+            y1 = x.Current;
+            Assert.Equal("NewKey2 NewValue2", y1.Key + " " + y1.Value);
+
+            environmentVariables.Add("NewKey3", "NewValue3");
+            //CopyTo
+            KeyValuePair<string, string>[] kvpa = new KeyValuePair<string, string>[10];
+            environmentVariables.CopyTo(kvpa, 0);
+            Assert.Equal("NewKey", kvpa[0].Key);
+            Assert.Equal("NewKey3", kvpa[2].Key);
+            Assert.Equal("NewValue3", kvpa[2].Value);
+
+            string[] kvp = new string[10];
+            Assert.Throws<ArrayTypeMismatchException>(() => { environmentVariables.CopyTo(kvp, 6); });
+            environmentVariables.CopyTo(kvpa, 6);
+            Assert.Equal("NewKey", kvpa[6].Key);
+            Assert.Equal("NewValue", kvpa[6].Value);
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => { environmentVariables.CopyTo(kvpa, -1); });
+
+            Assert.Throws<ArgumentException>(() => { environmentVariables.CopyTo(kvpa, 9); });
+
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                KeyValuePair<string, string>[] kvpanull = null;
+                environmentVariables.CopyTo(kvpanull, 0);
+            });
+        }
```

Comment:
Hey @stephentoub you are right, but now I realized that ProcessStartInfoTests are included for netstandard1.4 so I moved this test to a netstandard1.7 target. If I add just one test that covers both properties then we would lose coverage fot netstandard1.4 since EnvironmentVariables property is not defined in netstandard1.7